### PR TITLE
Parallelize SSRR requests for namespaces

### PR DESCRIPTION
**Pull request by jlpadilla**
Signed-off-by: Jorge Padilla <jpadilla@redhat.com>

**Related Issue:**  stolostron/backlog#22013

### Description of changes
- Use an async go routine for each namespace SSRR request.

### CODE REVIEW

File: pkg/rbac/userData.go [hunk 1]
```diff
@@ -277,10 +277,82 @@ func (user *UserDataCache) updateUserManagedClusterList(cache *Cache, ns string)
...
+		klog.V(9).Infof("SelfSubjectRulesReviews Kube API result for ns:%s : %v\n", ns, prettyPrint(result.Status))
+	}
+
+	// Process the SSRR result and add to this UserDataCache object.
+	for _, rules := range result.Status.ResourceRules {
+		for _, verb := range rules.Verbs {
+			if verb == "list" || verb == "*" {
+				for _, res := range rules.Resources {
+					for _, api := range rules.APIGroups {
+						lock.Lock()
```
Comment (Anxhela21) [inline pkg/rbac/userData.go:305]: Wondering would it be better to lock after we find resource with verb == "list" || verb ="*" (third line in loop)?

Comment (jlpadilla) [reply pkg/rbac/userData.go:305]: Maybe. I don't have a definite answer but here are my thoughts on the pros and cons.
- Here we lock for the shortest period of time. This creates a large number of lock/unlock cycles which allows other go routines to get the lock and continue their processing.
- Moving to the outside loop we have less lock/unlock cycles, which requires less computation, but the hold the lock for a longer period blocking other go routines.

Comment (Anxhela21) [reply pkg/rbac/userData.go:305]: these are good points. we can revisit later as potential optimization.

**Pull request merged**